### PR TITLE
IGNITE-20442 Sql. Extend grammar with transaction related statements.

### PR DESCRIPTION
**Pull request by lowka**
Updates SQL grammar to include `START TRANSACTION` and `COMMIT` statements.

https://issues.apache.org/jira/browse/IGNITE-20442

----

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCommitTransaction.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+/**
+ *  Parse tree for {@code COMMIT} statement.
+ */
+public class IgniteSqlCommitTransaction extends SqlCall {
+
+    /** Commit transaction operator. */
+    private static final SqlOperator OPERATOR = new SqlSpecialOperator("COMMIT", SqlKind.OTHER);
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCommitTransaction.java:35]: Why not ``SqlKind.COMMIT``?

Comment (lowka) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCommitTransaction.java]: I missed it. Fixed.

**Review by zstan — APPROVED** [review]

**Review by ygerzhedovich — APPROVED** [review]

**Comment by lowka** [comment]
Added start transaction mode that works similar to IgniteSqlIndexType.

**Review by ygerzhedovich — APPROVED** [review]

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl [hunk 2]
```diff
@@ -650,3 +650,37 @@ SqlTypeNameSpec IgniteDateTimeTypeName() :
...
+        <READ>
+        (
+          <ONLY> {
+            mode = IgniteSqlStartTransactionMode.READ_ONLY;
+          }
+          |
+          <WRITE> {
+            mode = IgniteSqlStartTransactionMode.READ_WRITE;
+          }
+        )
```
Comment (AMashenkov) [inline modules/sql-engine/src/main/codegen/includes/parserImpls.ftl:672]:
```suggestion
        <READ> <ONLY> {
            mode = IgniteSqlStartTransactionMode.READ_ONLY;
          }
          |
          <READ> <WRITE> {
            mode = IgniteSqlStartTransactionMode.READ_WRITE;
          }
```
Is it more readable now? 
I see `ColumnNameWithSortDirection` uses the same notation for "nulls first"

Comment (lowka) [reply modules/sql-engine/src/main/codegen/includes/parserImpls.ftl]: Thanks. Fixed.

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/SqlOtherStatementsTest.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import org.apache.calcite.sql.pretty.SqlPrettyWriter;
+import org.apache.ignite.lang.ErrorGroups.Sql;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+/**
+ * Tests for other SQL statements.
+ */
+public class SqlOtherStatementsTest {
```
Comment (zstan) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/SqlOtherStatementsTest.java:34]: Plz rename this class and change description a bit informative, people who not familiar with product usually search appropriate test by start typing keyword, "other" and "statement" is not suit for such a change.

**Review by zstan — APPROVED** [review]

**Pull request merged**